ckanext/protected_resources/plugin.py

```diff
@@ -68,7 +34,7 @@ def resource_delete(context, data_dict):
     # if you can update a package then you can delete it
     can_update = tk.check_access('package_update', context, pkg_dict)
     if can_update:
-        if resource.get('is_protected', None):
+        if resource_has_protected_status(resource['id']):
```

Comment:
2/10
Would prefer less nesting here:
```python
    if not can_update:
      return {'success': False, 'msg': 'Resource delete is not allowed'}
    if resource_has_protected_status(resource['id']):
      return {'success': False, 'msg': 'Contact a system administrator to delete this resource'}

    return {'success': True}
```

Reply:
2/10
Would the message be better as: `Contact a system administrator to un-lock this resource to allow deletion`?

Reply:
1/10
Below at line 50:
Do we want this `log.info` call still?

Reply:
Yes I think it makes sense to log that we are overriding the package_delete auth function, as it will give us some information if it disappears, due to another plugin overriding the same auth function, which will break the functionality of this plugin